monkestation/code/datums/components/irradiated.dm

```diff
@@ -0,0 +1,13 @@
+/datum/component/irradiated/RegisterWithParent()
+	. = ..()
+	if(!ismob(parent)) // no, you still have to go under the shower
+		RegisterSignal(parent, COMSIG_ATOM_EXPOSE_REAGENT, PROC_REF(on_expose_reagent))
+
+/datum/component/irradiated/UnregisterFromParent()
+	. = ..()
+	UnregisterSignal(parent, COMSIG_ATOM_EXPOSE_REAGENT)
+
+/datum/component/irradiated/proc/on_expose_reagent(atom/source, datum/reagent/reagent, reac_volume)
+	SIGNAL_HANDLER
+	if(istype(reagent, /datum/reagent/medicine/potass_iodide) && reac_volume >= 1)
+		qdel(src)
```

Comment:
You might have to call RemoveComponent() on parent here but I may be misremembering

Reply:
![image](https://github.com/Monkestation/Monkestation2.0/assets/65794972/dff2d6ad-444a-4904-ba75-49f4fec9e43e)

i saw other components that `qdel(src)` so i assume it's fine